### PR TITLE
Implement `IsMoveableVariable` check for inline array element access.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Relates to test plan https://github.com/dotnet/roslyn/issues/67826

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@tannergooding FYI

**Comment by AlekseyTs** [comment]
@cston, @jjonescz, @dotnet/roslyn-compiler Please review

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs [hunk 1]
```diff
@@ -2584,6 +2584,18 @@ internal bool IsMoveableVariable(BoundExpression expr, out Symbol accessedLocalO
...
                             continue;
                         }
+                    case BoundKind.InlineArrayAccess:
+                        {
+                            var elementAccess = (BoundInlineArrayAccess)expr;
+
+                            if (elementAccess.GetItemOrSliceHelper is WellKnownMember.System_Span_T__get_Item or WellKnownMember.System_ReadOnlySpan_T__get_Item)
+                            {
+                                expr = elementAccess.Expression;
+                                continue;
```
Comment (jjonescz) [inline src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs:2594]: Should the behavior be detailed in the speclet?

**Review by jjonescz — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs:2594]: > Should the behavior be detailed in the speclet?

https://github.com/dotnet/csharplang/pull/7362

**Pull request merged**